### PR TITLE
Dev/front

### PR DESCRIPTION
**Pull request by mohamedelhadi123**

### CODE REVIEW

File: services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public List<IdeaEntity> getPublishedIdeas(IdeaEntity.Status PUBLISHED){
+        TypedQuery<IdeaEntity> query = getEntityManager().createNamedQuery("Idea.getIdeasPublished", IdeaEntity.class);
+        query.setParameter("PUBLISHED",PUBLISHED);
+        return query.getResultList();
+
+
+
+    }
+
+    public List<IdeaEntity> getDRAFETIdeas(IdeaEntity.Status DRAFET){
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java]: To remove

File: services/src/main/java/org/exoplatform/ideation/dto/CommentDTO.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            this.createdTime= formatter.format(comment.getCreatedTime());
+        }
+        this.commentText = comment.getCommentText();
+        this.id_ideac = comment.getIdea().getId();
+    }
+    private Long id;
+    private String user;
+    private String createdTime;
+    private String  commentText;
+    private Long id_ideac;
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/dto/CommentDTO.java]:
```suggestion
    private Long ideaID;
```

File: services/src/main/java/org/exoplatform/ideation/dto/FavoritDTO.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package org.exoplatform.ideation.dto;
+
+import org.exoplatform.ideation.entities.FavoriteEntity;
+
+import java.io.Serializable;
+
+public class FavoritDTO implements Serializable {
+    private String user;
+    private Long id_Ideaf;
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/dto/FavoritDTO.java]:
```suggestion
    private Long ideaID;
```

File: services/src/main/java/org/exoplatform/ideation/dto/RatingDTO.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import org.exoplatform.ideation.entities.RatingEntity;
+
+import java.io.Serializable;
+
+public class RatingDTO implements Serializable {
+    public RatingDTO() {
+    }
+    private RatingEntity.Status status;
+    private String user;
+    private Long id_idear;
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/dto/RatingDTO.java]:
```suggestion
    private Long ideaID;
```

File: services/src/main/java/org/exoplatform/ideation/entities/CommentEntity.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+import javax.persistence.*;
+import java.io.Serializable;
+import java.util.Date;
+
+@Entity(name="comment")
+@ExoEntity
+@Table(name = "IDEATION_COMMENTS")
+@NamedQueries({
+        @NamedQuery(name = "Comment.getComment", query = "select c from comment c where c.idea.id=:id"),
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/entities/CommentEntity.java]: No need for this query, remove it

File: services/src/main/java/org/exoplatform/ideation/entities/FavoriteEntity.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+import javax.persistence.*;
+import java.io.Serializable;
+@Entity(name="fav")
+@ExoEntity
+@Table(name = "IDEATION_FAV")
+@NamedQueries({
+        @NamedQuery(name = "Fav.getAllFavByUser", query = "select f from fav f where f.user=:user"),
+        @NamedQuery(name = "Fav.getAllFavByUserAndId", query = "select f from fav f where f.user=:user AND f.idea.id=:id"),
+        @NamedQuery(name = "Fav.getAllFavById", query = "select f from fav f where f.idea.id=:id")
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/entities/FavoriteEntity.java]:
```suggestion
        @NamedQuery(name = "Fav.getAllFavByIdeaID", query = "select f from fav f where f.idea.id=:id")
```

File: services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+import javax.persistence.*;
+import java.io.Serializable;
+import java.util.Date;
+
+@Entity(name = "idea")
+@ExoEntity
+@Table(name = "IDEA_IDEAS")
+@NamedQueries({
+        @NamedQuery(name = "Idea.getIdeasByUser", query = "select i from idea i where i.user= :user"),
+        @NamedQuery(name = "Idea.getIdeasPublished", query = "select i from idea i where i.status= :PUBLISHED"),
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java]:
```suggestion
        @NamedQuery(name = "Idea.getIdeasByStatus", query = "select i from idea i where i.status= :status"),
```

File: services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        @NamedQuery(name = "Idea.getIdeasDRAFET", query = "select i from idea i where i.status= :DRAFET"),
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java]: To remove

File: services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+        @NamedQuery(name = "Idea.getIdeasARCHIVED", query = "select i from idea i where i.status= :ARCHIVED"),
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java]: To remove

File: services/src/main/java/org/exoplatform/ideation/dao/CommentImpDAO.java [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+import org.exoplatform.commons.persistence.impl.GenericDAOJPAImpl;
+import org.exoplatform.ideation.entities.CommentEntity;
+
+
+import javax.persistence.TypedQuery;
+import java.util.List;
+
+public class CommentImpDAO extends GenericDAOJPAImpl<CommentEntity,Long> {
+
+    public List<CommentEntity> getComment(Long id){
```
[the remark this answers is not in the document]
Comment (mohamedelhadi123) [reply services/src/main/java/org/exoplatform/ideation/dao/CommentImpDAO.java]: this "id" is used to retrieve data from the other table idea: it is a foreign key

File: services/src/main/java/org/exoplatform/ideation/entities/CommentEntity.java (hunk 5 above)
Comment (mohamedelhadi123) [reply services/src/main/java/org/exoplatform/ideation/entities/CommentEntity.java]: id idea used to show the commented idea

File: services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+public class IdeaImpDAO extends GenericDAOJPAImpl<IdeaEntity ,Long> {
+
+
+
+    public List<IdeaEntity> getIdeaByUser(String user){
+        TypedQuery<IdeaEntity> query = getEntityManager().createNamedQuery("Idea.getIdeasByUser", IdeaEntity.class);
+        query.setParameter("user", user);
+        return query.getResultList();
+    }
+    public List<IdeaEntity> getPublishedIdeas(IdeaEntity.Status PUBLISHED){
```
[the remark this answers is not in the document]
Comment (mohamedelhadi123) [reply services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java]: done

File: services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+    public List<IdeaEntity> getDRAFETIdeas(IdeaEntity.Status DRAFET){
+        TypedQuery<IdeaEntity> query = getEntityManager().createNamedQuery("Idea.getIdeasDRAFET", IdeaEntity.class);
+        query.setParameter("DRAFET",DRAFET);
+        return query.getResultList();
+
+
+    }
+
+    public List<IdeaEntity> getARCHIVEDIdeas(IdeaEntity.Status ARCHIVED){
```
[the remark this answers is not in the document]
Comment (mohamedelhadi123) [reply services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java]: done

File: services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+        TypedQuery<IdeaEntity> query = getEntityManager().createNamedQuery("Idea.getIdeasARCHIVED", IdeaEntity.class);
+        query.setParameter("ARCHIVED",ARCHIVED);
+        return query.getResultList();
+
+
+
+    }
+
+    public List<IdeaEntity> getPublishedIdeaByUser(IdeaEntity.Status PUBLISHED,String user){
```
[the remark this answers is not in the document]
Comment (mohamedelhadi123) [reply services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java]: done

File: services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java (hunk 1 above)
Comment (mohamedelhadi123) [reply services/src/main/java/org/exoplatform/ideation/dao/IdeaImpDAO.java]: done

File: services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java (hunk 8 above)
Comment (mohamedelhadi123) [reply services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java]: done

File: services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java (hunk 9 above)
Comment (mohamedelhadi123) [reply services/src/main/java/org/exoplatform/ideation/entities/IdeaEntity.java]: done

File: services/src/main/java/org/exoplatform/ideation/dao/CommentImpDAO.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (ahamdi) [reply services/src/main/java/org/exoplatform/ideation/dao/CommentImpDAO.java]: OK

File: services/src/main/java/org/exoplatform/ideation/entities/CommentEntity.java (hunk 5 above)
Comment (ahamdi) [reply services/src/main/java/org/exoplatform/ideation/entities/CommentEntity.java]: OK

**Pull request merged**